test/metabase/lib/binning_test.cljc

```diff
@@ -86,3 +87,31 @@
       true  {:strategy :bin-width, :bin-width 10, :num-bins 3} {:strategy :bin-width, :bin-width 10, :num-bins 4}
       true  {:strategy :bin-width, :bin-width 10}              {:strategy :bin-width, :bin-width 10, :metadata-fn (fn [] nil)}
       false {:strategy :bin-width, :bin-width 10}              {:strategy :bin-width, :bin-width 20})))
+
+(deftest ^:parallel binning-is-propagated-into-next-stage-display-name
+  (let [query (as-> meta/metadata-provider $
+                (lib/query $ (meta/table-metadata :orders))
+                (lib/aggregate $ (lib/count))
+                (lib/breakout $ (lib/with-binning
+                                 (meta/field-metadata :orders :total)
+                                 (m/find-first (comp #{"50 bins"} :display-name)
+                                               (lib/available-binning-strategies $ (meta/field-metadata :orders :total)))))
+                (lib.stage/append-stage $))]
+    (testing "Binning is present in next stage display-name"
+      (is (some? (m/find-first (comp #{"Total: 50 bins"} :display-name)
+                               (lib/visible-columns query)))))
+    (testing "Binning is present in next next stage display-name"
+      (is (some? (m/find-first (comp #{"Total: 50 bins"} :display-name)
+                               (lib/visible-columns
+                                (lib.stage/append-stage query))))))
+    (testing "Application of same binning on next stage does not modify display name"
+      (is (some? (m/find-first (comp #{"Total: 50 bins"} :display-name)
+                               (let [first-stage-binned-column (m/find-first (comp #{"Total: 50 bins"} :display-name)
+                                                                             (lib/visible-columns query))]
+                                 (lib/visible-columns
+                                  (as-> query $
+                                    (lib/breakout $ (lib/with-binning
+                                                     first-stage-binned-column
+                                                     (m/find-first (comp #{"50 bins"} :display-name)
+                                                                   (lib/available-binning-strategies
+                                                                    $ first-stage-binned-column)))))))))))))
```

Comment:
What about a different binning?

Reply:
I though that would be an overkill, as I've added new tests for that in `query_processor_test/binning_test.clj`. Considering that, do you think it should be added also here?